os-eap-launch/added/launch/jgroups.sh

```diff
@@ -21,22 +21,41 @@ function configureEnv() {
 function configure_jgroups_encryption() {
   jgroups_encrypt=""
 
-  if [ -n "${JGROUPS_ENCRYPT_SECRET}" ]; then
-    if [ -n "${JGROUPS_ENCRYPT_NAME}" -a -n "${JGROUPS_ENCRYPT_PASSWORD}" ] ; then
-      # For new JGroups we need to use SYM_ENCRYPT protocol
-      jgroups_encrypt="\
-        <protocol type=\"SYM_ENCRYPT\">\
-          <property name=\"provider\">SunJCE</property>\
-          <property name=\"sym_algorithm\">AES</property>\
-          <property name=\"encrypt_entire_message\">true</property>\
-          <property name=\"keystore_name\">${JGROUPS_ENCRYPT_KEYSTORE_DIR}/${JGROUPS_ENCRYPT_KEYSTORE}</property>\
-          <property name=\"store_password\">${JGROUPS_ENCRYPT_PASSWORD}</property>\
-          <property name=\"alias\">${JGROUPS_ENCRYPT_NAME}</property>\
-        </protocol>"
-    else
-      log_warning "Partial JGroups encryption configuration, the communication within the cluster WILL NOT be encrypted."
-    fi
-  fi
+  JGROUPS_ENCRYPT_PROTOCOL="${JGROUPS_ENCRYPT_PROTOCOL:-SYM_ENCRYPT}"
+
+  case "${JGROUPS_ENCRYPT_PROTOCOL}" in
+    "SYM_ENCRYPT")
+      if [ -n "${JGROUPS_ENCRYPT_SECRET}" -a -n "${JGROUPS_ENCRYPT_NAME}" -a -n "${JGROUPS_ENCRYPT_PASSWORD}" ] ; then
+        log_info "Configuring JGroups cluster traffic encryption protocol to SYM_ENCRYPT."
+        # For new JGroups we need to use SYM_ENCRYPT protocol
+        jgroups_encrypt="<protocol type=\"SYM_ENCRYPT\">\n\
+                        <property name=\"provider\">SunJCE</property>\n\
+                        <property name=\"sym_algorithm\">AES</property>\n\
+                        <property name=\"encrypt_entire_message\">true</property>\n\
+                        <property name=\"keystore_name\">${JGROUPS_ENCRYPT_KEYSTORE_DIR}/${JGROUPS_ENCRYPT_KEYSTORE}</property>\n\
+                        <property name=\"store_password\">${JGROUPS_ENCRYPT_PASSWORD}</property>\n\
+                        <property name=\"alias\">${JGROUPS_ENCRYPT_NAME}</property>\n\
+                    </protocol>"
+      else
+        log_warning "Partial JGroups encryption configuration, the communication within the cluster WILL NOT be encrypted."
```

Comment:
This warning will be always generated if user didn't set any environment regarding jgroups. Previous version printed warning only if SECRET envs was set and either USERNAMe or PASSWORD was missing. Is this what we want?

Reply:
@Crumby 
Good catch, Pavel! The image has been updated (see most recent version) that warning message to distinguish between the cases the JGroups configuration is partial, or not set at all (the cluster traffic wouldn't be encrypted in none of these cases), and based on that print the appropriate state.